realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -1381,14 +1381,28 @@ public void run() {
     /**
      * Removes all objects of the specified class.
      *
+     * DEPRECATED: Use {@link #delete(Class)} instead.
+     *
      * @param clazz the class which objects should be removed.
      * @throws IllegalStateException if the corresponding Realm is closed or in an incorrect thread.
      */
+    @Deprecated
     public void clear(Class<? extends RealmObject> clazz) {
+        delete(clazz);
+    }
+
+    /**
+     * Deletes all objects of the specified class from the Realm.
+     *
+     * @param clazz the class which objects should be removed.
+     * @throws IllegalStateException if the corresponding Realm is closed or called from an incorrect thread.
+     */
+    public void delete(Class<? extends RealmObject> clazz) {
```

Comment:
Swift uses `delete(Class)`, but I was debating the above myself as there are precedence for both, e.g. `Collection.removeAll(Collection)`. Personally I lean slightly towards using `delete(X)` for deleting any subset and `deleteAll()` for deleting the entire thing on either Realm/RealmList/RealmResults.